driver/src/price_finding/naive_solver.rs

```diff
@@ -0,0 +1,74 @@
+use crate::models;
+
+use super::error::PriceFindingError;
+use super::price_finder_interface::Solution;
+use web3::types::U256;
+use crate::models::Order;
+
+impl Order {
+    fn matches(&self, other: &Order) -> bool {
```

Comment:
How about having this return an `Option<enum>` which can be either TypeI-A TypeI-B or TypeII (maybe you can come up with some more descriptive names for them).

Then in your loop below you can use rusts awesome pattern matching feature.

Reply:
OMG yes! The hardest part will be coming up with the better names.